ale_linters/go/gobuild.vim

```diff
@@ -1,17 +1,126 @@
-" Author: dzhou121 <dzhou121@gmail.com>
+" Author: Joshua Rubin <joshua@rubixconsulting.com>
 " Description: go build for Go files
 
-function! s:FindGobuildScript() abort
-    return g:ale#util#stdin_wrapper . ' .go go build -o /dev/null'
+" inspired by work from dzhou121 <dzhou121@gmail.com>
+
+function! s:goenv()
+  if exists('s:go_env')
+    return s:go_env
+  endif
+
+  let l:env = systemlist('go env GOPATH GOROOT')
+
+  let s:go_env = {
+  \ 'GOPATH': l:env[0],
+  \ 'GOROOT': l:env[1],
+  \}
+
+  return s:go_env
+endfunction
+
+let s:splitchar = ':'
+if !has('unix')
+  let s:splitchar = ';'
+endif
+
+" get a list of all source directories from $GOPATH and $GOROOT
+function! s:srcdirs()
+  let l:env = s:goenv()
+  let l:paths = split(l:env.GOPATH, s:splitchar)
+  call add(l:paths, l:env.GOROOT)
+  return l:paths
```

Comment:
Could you add a single blank line before any `return` statement, `if` statement, or loop which doesn't begin a new level of indentation?